src/lib.rs

```diff
@@ -181,6 +186,89 @@ pub struct SystemHandle {
 pub struct Context<A: Actor + ?Sized> {
     pub system_handle: SystemHandle,
     pub myself: Addr<A>,
+    timer: TimerRef,
+}
+
+impl<A: Actor + ?Sized> Context<A> {
+    /// Sends a message once to the calling actor itself, after a delay.
+    /// Cancel with a call to `context.cancel_timer(schedule_token);`
+    pub fn send_once_to_self(&mut self, delay: Duration, msg: A::Message) -> ScheduleToken {
+        self.send_once_to::<A>(delay, msg, self.myself.recipient.clone())
+    }
```

Comment:
Public API wise, it feels somewhat unsymmetrical that `send()` is on `Recipient`, while `send_{once,recurring}*()` are on `Context`.
Do I get it right that it is because we didn't want to stick another field (`TimerRef`) to `Recipient`?

Reply:
Agree that `context.myself.send_once(...)` reads more symmetric. It allows calling the function outside the actor, too, whether it's a good idea or not. `TimerRef` seems lightweight enough to copy to each `Recipient`?

Reply:
I'll try this out! I think it makes sense and it's not super heavy on the recipient.

Reply:
So the immediate issue I run into here is: How do I construct an `Addr` or `Recipient` without involvement of the actor `System`?

Right now users of the crate can do something like:

```rust
let addr = Addr::default();
let another_addr = Addr::with_capacity(50);

let recipient: Recipient<()> = addr.recipient();
```

However there is no way to access a reference to the timer thread without passing it explicitly, which I think is a major hit on ergonomics. Am I missing anything?

Reply:
Yeah, we'll need to make it `System::addr()` and `System::addr_with_capacity()` I suppose.